model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -402,26 +427,104 @@ def _parse_outputs(self, metrics):
         ]:
             os.remove(perf_config['latency-report-file'])
 
-    def _extract_metrics_from_row(self, requested_metrics, row_metrics):
+    def _extract_metrics_from_row(self, requested_metrics: List[Record],
+                                  row_metrics: Dict[str, str]) -> List[Record]:
         """ 
         Extracts the requested metrics from the CSV's row and creates a list of Records
         """
-        perf_records = []
+        perf_records: List[Record] = []
+
+        perf_metrics, perf_values = self._find_requested_perf_metrics(
+            requested_metrics, row_metrics)
+        perf_records = self._add_perf_metrics_to_records(
+            perf_metrics, perf_values, perf_records)
+
+        gpu_metrics, gpu_values = self._find_requested_gpu_metrics(
+            requested_metrics, row_metrics)
+        perf_records = self._add_gpu_metrics_to_records(gpu_metrics, gpu_values,
+                                                        perf_records)
+
+        return perf_records
+
+    def _find_requested_perf_metrics(
+            self, requested_metrics: List[Record],
+            row_metrics: Dict[str, str]) -> Tuple[List[List[Any]], List[float]]:
+        perf_metrics: List[List[Any]] = []
+        values: List[float] = []
         for perf_metric in PerfAnalyzer.perf_metric_table:
-            if self._is_perf_metric_requested_and_in_row(
-                    perf_metric, requested_metrics, row_metrics):
-                value = float(row_metrics[perf_metric[PerfAnalyzer.CSV_STRING]]
-                             ) / perf_metric[PerfAnalyzer.REDUCTION_FACTOR]
+            if self._is_metric_requested_and_in_row(perf_metric,
+                                                    requested_metrics,
+                                                    row_metrics):
+                value = float(row_metrics[str(
+                    perf_metric[PerfAnalyzer.CSV_STRING])])
+                reduction_factor = float(
+                    str(perf_metric[PerfAnalyzer.REDUCTION_FACTOR]))
+                perf_value = value / reduction_factor
+
+                perf_metrics.append(perf_metric)
+                values.append(perf_value)
+
+        return (perf_metrics, values)
+
+    def _add_perf_metrics_to_records(
```

Comment:
I don't like perf_records being an input and an output. You have two functions that return a list of records. You don't have to have the appending hidden in the function.

perf_records = extract_perf_records_from_row
gpu_records = extract_gpu_records_from_row
combined_records = perf_records + gpu_records
return combined_records

Reply:
Done